GoMeet/src/main/webapp/js/interactive-map.js

```diff
@@ -110,24 +109,31 @@ async function fetchLocations(map) {
 
 /** Builds a HTML element to display the location's data and a vote button. */
 function buildInfoWindowVote(title, voteCount, note, keyString) {
-  let titleContainer = document.createElement('span');
-  titleContainer.setAttribute('id', 'displayTitle');
-  titleContainer.innerText = title;
-
-  let noteContainer = document.createElement('span');
-  noteContainer.setAttribute('id', 'displayNote');
-  noteContainer.innerText = note;
+  const titleContainer = createSpanContainer(title, 'displayTitle');
+  const noteContainer = createSpanContainer(note, 'displayNote');
+  const voteContainer = createSpanContainer(voteCount, 'displayVoteCount');
 
   const button = document.createElement('button');
+  button.setAttribute('id', 'voteButton');
   button.appendChild(document.createTextNode('VOTE'));
   button.onclick = () => {
     MeetingLocationDAO.updateLocation(keyString);
+    const currVote = voteContainer.innerText;
+    voteContainer.innerText = parseInt(currVote) + 1;
```

Comment:
The code is directly updating the HTML display voteCount, which could be dangerous in that if the `MeetingLocationDAO.updateLocation()` call fails to update (error occurs during update), then the voteCount in the UI is inconsistent with the voteCount in the server. 

Will this be refactored later on to re-render the page to update the voteCount instead of directly updating it?

Reply:
Yep, that's a good point. I do plan to refactor but instead I plan to make updateLocation() check the status of the response. If it is not 200 then updateLocation() will throw an error. In the button.onclick() it would look like this:

```
try {
    MeetingLocationDAO.updateLocation(keyString);
    const currVote = voteContainer.innerText;
    voteContainer.innerText = parseInt(currVote) + 1;
} catch (error) {
  handleError();
}
```

That way the innerText is only updated if the updateLocation() does not throw an error.

Reply:
Yep sounds good! I think its okay for this PR, but optionally, if you want to be extra safe, you could just not update the votes here, and have the user refresh the page to see the updated votes.